util/pace/util/grid/stretch_transformation.py

```diff
@@ -0,0 +1,129 @@
+import copy as cp
```

Comment:
We use `cp` strictly for `import cupy as np`, I suggest you just `import copy` and use `copy.X` everywhere.